metaflow/graph.py

```diff
@@ -256,3 +250,57 @@ def node_specs():
                 self, nodes="\n".join(node_specs()), edges="\n".join(edge_specs())
             )
         )
+
+    def output_steps(self):
+
+        steps_info = {}
+        steps_structure = []
+
+        def node_to_dict(name, node):
+            return {
+                "name": name,
+                "type": node.type,
```

Comment:
let's map types to something more sensible since the internal names are not super clear, e.g.
```
TYPE_MAP = {
   'foreach': 'split-foreach',
   'split': 'split-static',
   'linear': 'linear',
   'end': 'end',
   'join': 'join'
}